### PR TITLE
feat: Fuzzy testing

### LINKED ISSUE #263: Fuzzy testing
Add a fuzzy testing framework and tests.

### PR DESCRIPTION
**Pull request by hhamud**
* Closes  #263

## Checklist
- [x] Linked to Github Issue
- [ ] Unit tests added
- [ ] Integration tests added.
- [x] This change requires new documentation.
  - [x] Documentation has been added/updated.

### CODE REVIEW

**Comment by hhamud** [comment]
I can't seem to even build the crate as cairo-lang-runner couldn't compile

**Review by igaray — APPROVED** [review]

File: Makefile [hunk 1]
```diff
@@ -143,3 +143,6 @@ runtime:
 
 runtime-ci:
 	cargo b --profile ci --all-features -p cairo-native-runtime && cp target/ci/libcairo_native_runtime.a .
+
+fuzz: check-llvm needs-cairo2 build-alexandria runtime-ci
+	cargo +nightly fuzz run fuzz-sierra
```
Comment (pefontana) [inline Makefile:148]: Hi @hhamud !
Is it necessary to use +nightly?
Can we avoid it for the moment?

**Review by pefontana — CHANGES_REQUESTED** [review]

File: Makefile (hunk 1 above)
Comment (hhamud) [reply Makefile:148]: its required unfortunately.

Comment (hhamud) [reply Makefile:148]: @pefontana is `+nightly` still an absolute blocker? cargo fuzzer requires it though I am not sure if honggfuzz does.

**Pull request closed without merging**